### PR TITLE
#184651060 Login Functionality with Email and Password

### PR DESCRIPTION
**Pull request by Edcreation**
#### What does this PR do?
- Login Route and Protected routes
- Deleted Unnecessary index.js
- Also changed the PR template 
- Tested and mocked login with email and password
- Added needed dependencies in readme
#### Description of Task to be completed?
- Login functionality with email and password
- Users cannot access the application without logging in without providing an email and password combination.
#### How should this be manually tested?
- use postman with post request to `http://localhost:{PORT}/users/login`
- use `http://localhost:{PORT}/users/protected-route` to check protected route
- content-type *application/json*
- fields are `email` and `password`
#### What are the relevant pivotal tracker/Trello stories?
- #184651060

### CODE REVIEW

**Review by AdrineUWERA — CHANGES_REQUESTED** [review]
The tags used in the swagger documentation aren't the same. I suggest we have a common tag for the authentication routes. Change the tag for the signup route to Authentication.

**Review by ishimwe5555 — CHANGES_REQUESTED** [review]
Could you use Redis with the passport.js file?

File: src/docs/apidoc.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const options = {
   },
   servers: [
     {
-      url: process.env.SWAGGER_URL // Port Number on this URL must match Server Port Number
+      url: process.env.SWAGGER_URL, // Port Number on this URL must match Server Port Number
     },
     {
       url: 'http://localhost:3000/',
```
Comment (Kress2000) [inline src/docs/apidoc.js:15]: @Edcreation , what if here, one changes the App PORT to another number, not 3000? I suggest here you use  this: 
```
url: `http://localhost:${process.env.PORT}/`
 ``` 
:  just to be flexible with whatever port used!

**Review by Kress2000 — CHANGES_REQUESTED** [review]

File: src/docs/apidoc.js (hunk 1 above)
Comment (Edcreation) [reply src/docs/apidoc.js:15]: Working on it 🚀

Comment (Edcreation) [reply src/docs/apidoc.js]: > @Edcreation , what if here, one changes the App PORT to another number, not 3000? I suggest here you use this:
> 
> ```
> url: `http://localhost:${process.env.PORT}/`
> ```
> 
> : just to be flexible with whatever port used!

Done!🚀

**Review by AdrineUWERA — APPROVED** [review]

File: src/index.js [hunk 2]
```diff
@@ -1,31 +1,40 @@
...
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 
 const specs = swaggerJSDoc(options);
+app.use('/myapi', swaggerUi.serve, swaggerUi.setup(specs));
+
+sequelize.sync();
 
 app.use('/api-doc', swaggerUi.serve, swaggerUi.setup(specs));
 app.use('', blogRoutes);
```
Comment (Pacific911) [inline src/index.js:31]: You can  remove this blogroute cause we won't use it anymore.

**Review by Pacific911 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (ishimwe5555) [reply src/index.js:31]: And as earlier requested we changed swagger route from /myapi  to /api-docs

Comment (Edcreation) [reply src/index.js]: > You can remove this blogroute cause we won't use it anymore.



> And as earlier requested we changed swagger route from /myapi to /api-docs

@ishimwe5555 ✅ changed `swagger route from /myapi to /api-docs`
@Pacific911 ✅ removed the blogroute

**Review by Kress2000 — APPROVED** [review]

**Review by ishimwe5555 — APPROVED** [review]

File: src/database/seeders/20230310115577-demo-user.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
+/* eslint-disable import/no-extraneous-dependencies */
 const { v4: uuidv4 } = require('uuid');
```
Comment (munezerobagira) [inline src/database/seeders/20230310115577-demo-user.js:2]: Can you use es6 syntax for this file as well

File: src/docs/apidoc.js [hunk 4]
```diff
@@ -4,34 +4,20 @@ const options = {
...
-      url: process.env.SWAGGER_URL // Port Number on this URL must match Server Port Number
+      url: `http://localhost:${process.env.PORT}/`,
+      description: 'Api Local server',
     },
     {
-      url: 'http://localhost:3000/',
-      description: 'Api server',
+      url: ` https://team-sostene-e-commerce-bn.onrender.com/`,
+      description: 'Api Hosted server',
     },
```
Comment (munezerobagira) [inline src/docs/apidoc.js:18]:
```suggestion
    },
    {
       url:`/`,
       description:"For calling server itself wether local or hosted"
    }
```

**Review by munezerobagira — CHANGES_REQUESTED** [review]

**Review by Pacific911 — APPROVED** [review]

File: src/database/migrations/20230322094623-create-user.js
Comment (princoo) [inline src/database/migrations/20230322094623-create-user.js:1]: this fields may need  `allowNull:false` property

Comment (Edcreation) [reply src/database/migrations/20230322094623-create-user.js:1]: > this fields may need `allowNull:false` property

Done!

File: src/database/seeders/20230310115577-demo-user.js (hunk 3 above)
Comment (Edcreation) [reply src/database/seeders/20230310115577-demo-user.js]: > Can you use es6 syntax for this file as well

Done!

File: src/docs/apidoc.js (hunk 4 above)
Comment (Edcreation) [reply src/docs/apidoc.js:14]: > 

Changed!

**Review by Kress2000 — APPROVED** [review]

**Review by JosephNgabo — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]
LGTM!!

**Review by Pacific911 — APPROVED** [review]
LGTM 👍

**Pull request merged**